boards/shields/nrf7002ek/nrf7002ek_coex.overlay

```diff
@@ -0,0 +1,16 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	nrf_radio_coex: nrf7002-coex {
+		status = "okay";
+		compatible = "nordic,nrf700x-coex";
+		status0-gpios = <&arduino_header 8 GPIO_ACTIVE_HIGH>;       /* D2 */
+		req-gpios = <&arduino_header 9 GPIO_ACTIVE_HIGH>;   /* D3 */
+		grant-gpios = <&arduino_header 10 (GPIO_PULL_DOWN | GPIO_ACTIVE_LOW)>;    /* D4 */
+		swctrl1-gpios = <&arduino_header 12 GPIO_ACTIVE_HIGH>;  /* D6 */
+		srrf-switch-gpios = <&arduino_header 14 GPIO_ACTIVE_HIGH>;  /* D8 */
```

Comment:
Is it nordic specific or should it be a shield variant?

Reply:
Are you referring to the co-ex entries? They are nRF70 specific.

Reply:
If you are asking about how these are included, please see an example https://github.com/zephyrproject-rtos/zephyr/blob/af7e11bb0a95bb48703b0ee8793736214d847aff/boards/shields/nrf7002ek/boards/nrf5340dk_nrf5340_cpuapp.overlay#L6. The applicability of these depends on the host, so, `boards` folder handles this.

Reply:
If [boards/shields/nrf7002ek/nrf7002ek_coex.overlay](https://github.com/zephyrproject-rtos/zephyr/pull/75714/files#diff-bcc4df48529ef6df9a5264f601cb16de3881f3fda7f3908985ef77fde418f5cd) can be used with any board then it should be a shield variant.

Reply:
But it cannot be used independently, it should always be used in conjunction with nrf7002ek.

Reply:
Add `#include "nrf7002ek.overlay"` to nrf7002ek_coex.overlay

Add
```
config SHIELD_NRF7002EK_COEX
	def_bool $(shields_list_contains,nrf7002ek_coex)
```
to Kconfig.shield

Reply:
> Add #include "nrf7002ek.overlay" to nrf7002ek_coex.overlay

Co-EX is a sub-feature on Wi-Fi, and a user wouldn't know when to use this shield and if it's even applicable for that host, the pins in the `nrf7002ek_coex.overlay` are decided based on applicability to the host, so, current approach handles this automatically using `boards` folder without any user intervention.

Reply:
Make it a shield variant, and describe in the doc when to use which variant.

Reply:
Nothing is handled automatically. You hardcoded this overlay to only be included for Nordic boards, adding three configuration overlays for no reason, but claimed that it is generic and vendor independent. Also, this overlay does not contain anything board specific, it can be applied to any board with an arduino header. Please do it right as a shield variant.

Reply:
> but claimed that it is generic and vendor independent.

Not really,  I was only talking about user not having to explicitly specify co-ex GPIOs and the developer of a particular board decides it's usage. So, if a customer is adding a new board, then he can decide to include that coex overlay file or not (which is what using a shield does), and it's hidden from the users. 

> Please do it right as a shield variant.

* `SHIELD =nrf7002_coex`: User decides co-ex GPIOs inclusion per-build
* `#include "../nrf7002ek_coex.overlay"` Developer decides co-ex GPIOs inclusion *once*

Reply:
> > but claimed that it is generic and vendor independent.
> 
> Not really, I was only talking about user not having to explicitly specify co-ex GPIOs and the developer of a particular board decides it's usage. So, if a customer is adding a new board, then he can decide to include that coex overlay file or not (which is what using a shield does), and it's hidden from the users.

The developer of the board or the customer who adds a new board (where?) decides it usage and adds new config overlay to upstream shield board directory? This nonsense, you have two options, do it right or close this PR. My -1 stays.

Reply:
> The developer of the board or the customer who adds a new board (where?) 

Can be upstream or a local change.

> adds new config overlay to upstream shield board directory?

If the board is upstream then yes, else this can be done locally too.

> This nonsense, you have two options, do it right or close this PR. My -1 stays.

I understand what you are trying to say about customer manually modifying `shield` for a new board, but on the flipside if SR Co-ex feature is enabled but shield isn't passed then build will fail (GPIOs are not passed) and then user should read the docs and figure out the passing of a special shield (not really shield, but an overlay wrapped under a shield)

Anyways, given the ultimatum, will move this to a `nrf7002_coex` shield.

Reply:
> but on the flipside if SR Co-ex feature is enabled

Is it a Kconfig option and where can I find it?

Reply:
Yes, see https://github.com/krish2718/zephyr/blob/add_7002ek/drivers/wifi/nrfwifi/Kconfig.nrfwifi#L166 it's used in the driver.